### ISSUE TITLE
windows: support for suspending jobs

### ISSUE THREAD

**Issue by rbrewer123** [scopatz/xonsh#236]
xonsh running on Windows allows use of the '&' operator in subprocess mode to create background processes.  These may be brought to the foreground via the 'fg' command.  ctrl-z to suspend the foreground process is currently not implemented.  

@wrywerytwreywery has stated that the 'psutils' package may provide some underlying Python tools for implementing process suspension on Windows.  I believe that msysGit bash allows suspending processes, so there must be some way to implement or emulate this feature on Windows.

**Comment by wrywerytwreywery** [comment]
`psutils` has support for suspending processes, so the actual suspending should be possible.  The hard part is getting the ^Z command to the process, I think (iirc, Windows treats ^Z as EOF, so I'm not sure how possible it is to grab that when a process is running).

**Comment by scopatz** [comment]
Maybe ^Z is terminal dependent too....

**Comment by wrywerytwreywery** [comment]
I think there's a chance, but I did a quick test in git bash, and that also didn't interpret ^Z.  I think the odds are slim that there is an easy way to add such a hook.

**Comment by gforsyth** [comment]
I'm pretty sure this is available in the current release.  Please re-open if it isn't working correctly.

**Comment by melund** [comment]
Nope. This hasn't been fixed yet. Not sure if there will ever be a bullet-proof solution. But there are some possibilities. See https://github.com/scopatz/xonsh/pull/962#issuecomment-222971467

**Comment by astronouth7303** [comment]
This is implemented in slug and will be fixed by #2337.
